src/behavior.js

```diff
@@ -92,7 +92,9 @@ export default class Behavior extends React.PureComponent {
   goTo = (key, config = {}) => {
     const isSequence = Array.isArray(key)
 
-    const { config: defaultConfig, state } = this.props
+    const { config: defaultConfig, state, freeze } = this.props
+
+    if (freeze) this.key = key
```

Comment:
This is not needed it's already done here https://github.com/ExodusMovement/haraka/blob/88d79c6065c62db5d5cf2e8454a8a695e16bd2c1/src/behavior.js#L140

Reply:
My intention here is to avoid executing the animation setup which is not needed on freeze mode.
 
Or... I can refactor it like the following to clearly separate on how we handle freeze and non-freeze. How do you think?

```
componentWillReceiveProps(nextProps) {
    const { currentState, disabled, freeze } = this.props
    const { currentState: nextCurrentState, disabled: nextDisabled } = nextProps

    if (currentState !== nextCurrentState) {
      if (freeze) this.key = nextCurrentState
      else this.goTo(nextCurrentState)
    }
```

Reply:
Yes, but this line already exists here https://github.com/ExodusMovement/haraka/blob/88d79c6065c62db5d5cf2e8454a8a695e16bd2c1/src/behavior.js#L140 for all the cases not just `freeze` so there is duplication here, unless I'm missing something

Reply:
Remove `if (freeze) this.key = key` and test, you should get the same result `this.key = key` is executed few lines below

Reply:
@nafisholeh as Ain asked why do you need to set the `key` on freeze, if it is already set in https://github.com/ExodusMovement/haraka/blob/ae2c2eab53db9f054bed959e84f091701312e7d0/src/behavior.js#L142

Reply:
@sonaye @andrejborstnik Apologies for the late response, was focusing on other tickets on current sprint.

My bad here, I forgot to add `return` statement 😅 
```suggestion
    if (freeze) {
      this.key = key
      return 
    }
```

This wont execute the rest of code on `goTo` function. Moreover, transition between page feels much snappier when I add this `return` statement.

Reply:
I made an update here [4aeecab](https://github.com/ExodusMovement/haraka/pull/16/commits/4aeecab4fe824e6b33dc227007bd007c09b4a19f)

Reply:
That looks fine to me. Do we not need `goTo` for non-animated transition? If it works as expected that is great!

Reply:
non-animated still need `this.key` to be updated with current state, and I see `goTo` is a means to update state whether it is animated or non-animated.
Additionally, in the future, if `goTo` is called from another place due to some circumstances, then it would be much easier for other devs to update state without worrying for non-animated/animated.